runtime/darwinia/src/migrations.rs

```diff
@@ -23,90 +27,29 @@ impl OnRuntimeUpgrade for CustomOnRuntimeUpgrade {
 }
 
 fn migrate() -> Weight {
-	let storages: &[(&[u8], &[&[u8]])] = &[
-		// TODO: check if we have left some balances locks
-		(
-			b"EcdsaRelayAuthority",
-			&[
-				b"Candidates",
-				b"Authorities",
-				b"NextAuthorities",
-				b"NextTerm",
-				b"AuthoritiesToSign",
-				b"MmrRootsToSignKeys",
-				b"MmrRootsToSign",
-				b"SubmitDuration",
-			],
-		),
-		(
-			// FrameV1 name.
-			b"DarwiniaEthereumRelay",
-			&[
-				b"ConfirmedHeaderParcels",
-				b"ConfirmedBlockNumbers",
-				b"BestConfirmedBlockNumber",
-				b"ConfirmedDepth",
-				b"DagsMerkleRoots",
-				b"ReceiptVerifyFee",
-				b"PendingRelayHeaderParcels",
-			],
-		),
-		// TODO: check if we have left some balances locks
-		(
-			// FrameV1 name.
-			b"Instance1DarwiniaRelayerGame",
-			&[
-				b"RelayHeaderParcelToResolve",
-				b"Affirmations",
-				b"BestConfirmedHeaderId",
-				b"RoundCounts",
-				b"AffirmTime",
-				b"GamesToUpdate",
-				b"Stakes",
-				b"GameSamplePoints",
-			],
-		),
+	Scheduler::migrate_v2_to_v3();
+
+	for precompile in DarwiniaPrecompiles::<Runtime>::used_addresses() {
+		EVM::create_account(&precompile, vec![0x60, 0x00, 0x60, 0x00, 0xFD]);
+	}
+
+	let removed_items: &[(&[u8], &[&[u8]])] = &[
 		(
-			b"EthereumBacking",
+			b"ToCrabBacking",
 			&[
-				b"TokenRedeemAddress",
-				b"DepositRedeemAddress",
-				b"SetAuthoritiesAddress",
-				b"RingTokenAddress",
-				b"KtonTokenAddress",
-				b"RedeemStatus",
-				b"LockAssetEvents",
+				b"SecureLimitedPeriod",
+				b"SecureLimitedRingAmount",
+				b"TransactionInfos",
+				b"RemoteMappingTokenFactoryAccount",
 			],
 		),
+		(b"KtonTreasury", &[b"ProposalCount", b"Proposals", b"Approvals"]),
```

Comment:
Are there existing KTON balances in the KTONTreasury pallet account, do we need to burn them?

Reply:
Yes. But let's leave them there.

RING and KTON are using the same pallet account.